test/plugins/collapseGroups.17.svg.txt

```diff
@@ -0,0 +1,43 @@
+Don't collapse group with a single child unless all attributes can be moved to the child.
+
+===
+
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <defs>
+        <linearGradient xlink:href="#a" id="c" x1="3.97" x2="3.97" y1=".348" y2="4.34" gradientUnits="userSpaceOnUse"/>
+    </defs>
+    <filter id="b">
+        <feGaussianBlur stdDeviation="1"/>
+    </filter>
+    <symbol viewBox="-9.5 -7 19 14">
+        <g style="filter:url(#b)">
+            <linearGradient id="a" x1="3.97" x2="3.97" y1="2.425" y2="7.55" gradientUnits="userSpaceOnUse">
+                <stop offset="1" style="stop-color:#fff"/>
+            </linearGradient>
+        </g>
+    </symbol>
+    <g transform="matrix(0 -1.2285 -1.2285 0 112.337 268.327)" style="filter:url(#b)">
+        <path d="M.84.348H7.1V5.8H3.344z" style="fill:url(#c)"/>
+    </g>
+</svg>
```

Comment:
I somewhat-strongly dislike this test case. It doesn't render in your browser, and it has too much cruft. Use something like this as the input instead.
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" width="60" height="60">
  <defs>
    <filter id="a">
      <feGaussianBlur stdDeviation="1"/>
    </filter>
  </defs>
  <g transform="scale(2)" style="filter:url(#a)">
    <rect width="20" height="20" fill="blue" style=""/>
  </g>
</svg>
```

Reply:
I'd like to keep it as is. The test file is a distillation of the Oxygen icon that was failing. It took me a really long time to isolate the problem, and I know for sure that the original file I submitted was failing before the change.

Reply:
Then I hope you will consider at least making it render in a browser and ideally distill it more

Reply:
Simplified the test case. New test case fails on main branch, passes here.

Reply:
The "incorrect" behavior renders the same as the "correct" behavior, but for the purposes of demonstrating the required behavior this test works fine.

Reply:
Yeah, I think it's pretty hard to build an example where the bug really makes a difference. That's one of the reasons it was so hard to track down. And also why the regression tests with "real" SVGs are important.

Reply:
```svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 88 88">
  <filter id="a">
    <feGaussianBlur stdDeviation="1"/>
  </filter>
  <g transform="matrix(0.6875,0,0,0.6875,20.34375,66.34375)" style="filter:url(#a)">
    <path d="M 33.346591,-83.471591 L -10.744318,-36.471591 L -10.49989,-32.5 L 34.409091,-77.375 L 78.951759,-33.318072 L 80.5625,-36.471591 L 35.471591,-83.471591 L 33.346591,-83.471591 z" style="fill-opacity:1"/>
  </g>
</svg>
```

Here is an example that reproduces the problem and clearly demonstrates the change in rendering caused by the plugin.

It looks like what you describe to be the issue and solution is correct and applies to this example. Could you use this for the test case though, please!

Reply:
I changed the test file.

I did pretty much what you did to isolate this, but this one was difficult to isolate because the file didn't fail if you only ran collapseGroups - you also had to run some other combination of plugins that would reorder the attributes in a way that would cause collapseGroups to trigger the error, and deleting parts of the original file sometimes changed things in such a way that the error didn't show up, so there were a lot of red herrings.

Also, see PR #1932 - this is a start on changing regression.js to make testing/analysis easier. I've been using this version to analyze problems and test changes. Still more to be done obviously, but it's a start.